modules/integration/src/test/scala/scala/cli/integration/BloopTests.scala

```diff
@@ -168,4 +168,25 @@ class BloopTests extends ScalaCliSuite {
         }
       }
     }
+
+  test("run bloop with jvm version if > 17") {
+    val hello = "Hello from Java 21"
+    val inputs = TestInputs(
+      os.rel / "Simple.java" ->
+        s"""|//> using jvm 21
+            |//> using javacOpt --enable-preview -Xlint:preview
+            |//> using javaOpt --enable-preview
+            |//> using mainClass Simple
+            |
+            |void main() {
+            |    System.out.println("$hello");
+            |}""".stripMargin
+    )
+    inputs.fromRoot { root =>
+      val res = runScalaCli("Simple.java").call(cwd = root)
```

Comment:
I'd first explicitly force Bloop to run on Java 17 here, to make sure nothing else is picked from the environment, some other test or w/e. And only then check if the directive ups Bloop to 21.